src/library.cpp

```diff
@@ -120,6 +120,25 @@ void Library::save()
     m_library.writeBookmarksToFile(kiwix::appendToDirectory(m_libraryDirectory.toStdString(), "library.bookmarks.xml"));
 }
 
+void Library::loadMonitorDir(QString dir)
+{
+    auto manipulator = LibraryManipulator(this);
+    auto manager = kiwix::Manager(&manipulator);
+    std::vector<std::string> paths{kiwix::getDataDirectory() + "/" + "library.xml"};
+    manager.reload(paths);
+    QDirIterator it(dir, {"*.zim"}, QDir::Files);
+    QString fullBookPath;
+    while (it.hasNext()) {
+        fullBookPath = it.next();
+        if(!manager.addBookFromPath(fullBookPath.toStdString())) {
+            qWarning()<<"Couldn't add "<<fullBookPath<<" in the library.";
```

Comment:
When a new ZIM file appears in MonitorDir, isn't this warning printed for all books that were present in the library before that?

Reply:
It doesn't print warnings. I kind of assumed addBookFromPath would already check for existing book

Reply:
I was confused here, but still my feeling that something is wrong remains. In such an implementation, whenever a new ZIM file appears, all books are re-read. @kelson42 is that acceptable?

Reply:
@veloman-yunkan @juuz0 No, if this can be avoided. Opening a ZIM file (and its Xapian indexes) is not a cheap operation. Therefore we should only do the necessary.

@veloman-yunkan You already had a similar challenge with monitoring a library in kiwix-serve. Maybe you can give a hint or two to @juuz0 here about the best path to follow.

Reply:
QFileSystemWatcher doesn't provide any other useful information

Comparing directory files before and after any changes in directory is another way. Not sure about any other

Reply:
> @veloman-yunkan You already had a similar challenge with monitoring a library in kiwix-serve. Maybe you can give a hint or two to @juuz0 here about the best path to follow.

Monitoring a library XML file is different from monitoring a directory for new or removed ZIM files. @juuz0 already knows how to do it:

> Comparing directory files before and after any changes in directory is another way.

Reply:
@juuz0 If I understand properly, QFileSystemWatcher is not able to tell you what has changed, which files have been removed, updated or even deleted. So a kind of diff will be needed. For performance reason this should be achieved without having to read the files themself, so each file shoudl be listed and identified based in filename, size and modification tine. The a diff shoukd be done. Without been prescriptive, a subclassing seems apropriate here,